### PR TITLE
Adding MOFED Install test

### PR DESCRIPTION
**Pull request by narasimhan-v**
This test verifies the installation of MOFED iso with different
combinations of input parameters.

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
[job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/494683/job_log.txt)

File: io/net/infiniband/mofed_install_test.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        Uninstalls MOFED, if installed fine.
+        """
+        cmd = "etc/init.d/openibd restart"
+        if not process.system(cmd, ignore_status=True, shell=True):
+            return
+        cmd = "ibstat"
+        if not process.system(cmd, ignore_status=True, shell=True):
+            return
+        cmd = './uninstall.sh --force'
+        if process.system(cmd, ignore_status=True, shell=True):
```
Comment (PraveenPenguin) [inline io/net/infiniband/mofed_install_test.py]: here also can you please `if process.system('./uninstall.sh --force', ignore_status=True, shell=True):` ?

File: io/net/infiniband/mofed_install_test.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    def uninstall(self):
+        """
+        Uninstalls MOFED, if installed fine.
+        """
+        cmd = "etc/init.d/openibd restart"
+        if not process.system(cmd, ignore_status=True, shell=True):
+            return
+        cmd = "ibstat"
+        if not process.system(cmd, ignore_status=True, shell=True):
```
Comment (PraveenPenguin) [inline io/net/infiniband/mofed_install_test.py]: here as well if not `process.system('ibstat', ignore_status=True, shell=True): ` please

File: io/net/infiniband/mofed_install_test.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        """
+        self.iso_location = self.params.get('iso_location', default='')
+        if self.iso_location is '':
+            self.skip("No ISO location given")
+        self.option = self.params.get('option', default='')
+        self.tarball = self.fetch_asset(self.iso_location, expire='10d')
+        cmd = "mount -o loop %s %s" % (self.tarball, self.srcdir)
+        process.run(cmd, shell=True)
+        self.pwd = os.getcwd()
+        os.chdir(self.srcdir)
```
Comment (narasimhan-v) [inline io/net/infiniband/mofed_install_test.py]: This is needed, as we need to execute the command from this location.

File: io/net/infiniband/mofed_install_test.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        cmd = './mlnxofedinstall %s --force' % self.option
+        if process.system(cmd, ignore_status=True, shell=True):
+            self.fail("Install Failed with %s" % self.option)
+
+    def uninstall(self):
+        """
+        Uninstalls MOFED, if installed fine.
+        """
+        cmd = "etc/init.d/openibd restart"
+        if not process.system(cmd, ignore_status=True, shell=True):
```
Comment (narasimhan-v) [inline io/net/infiniband/mofed_install_test.py]: I am using the variable `cmd` only everywhere. So, I do not think it is a problem

File: io/net/infiniband/mofed_install_test.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+    def setUp(self):
+        """
+        Mount MOFED iso.
+        """
+        self.iso_location = self.params.get('iso_location', default='')
+        if self.iso_location is '':
+            self.skip("No ISO location given")
+        self.option = self.params.get('option', default='')
+        self.tarball = self.fetch_asset(self.iso_location, expire='10d')
```
Comment (narasimhan-v) [inline io/net/infiniband/mofed_install_test.py]: Sure, will change it.

File: io/net/infiniband/mofed_install_test.py (hunk 3 above)
Comment (PraveenPenguin) [inline io/net/infiniband/mofed_install_test.py]: sure can you use this change dir in other helper  function  as there is no command executed after change dir

```
       self.option = self.params.get('option', default='')
        self.tarball = self.fetch_asset(self.iso_location, expire='10d')
        cmd = "mount -o loop %s %s" % (self.tarball, self.srcdir)
        process.run(cmd, shell=True)
        self.pwd = os.getcwd()
        os.chdir(self.srcdir)

def install(self):
```

File: io/net/infiniband/mofed_install_test.py (hunk 4 above)
Comment (PraveenPenguin) [inline io/net/infiniband/mofed_install_test.py]: sure that make sense but still why use at least once to make garbage collector to do the job 
:)

Comment (PraveenPenguin) [inline io/net/infiniband/mofed_install_test.py]: but again it your's call it is just a suggestion

Comment (narasimhan-v) [inline io/net/infiniband/mofed_install_test.py]: I agree with your point, but I prefer this way where each command is assigned to a variable :-)

File: io/net/infiniband/mofed_install_test.py (hunk 3 above)
Comment (narasimhan-v) [inline io/net/infiniband/mofed_install_test.py]: Agreed, will push this to install()

**Pull request closed without merging**